src/USER-SCAFACOS/fix_scafacos.cpp

```diff
@@ -0,0 +1,454 @@
+
+  // TODO: for now disable short-range calculations within LAMMPS
+  near_field_flag = 0;
+
+  // enter all parameters required to ScaFaCoS handle
+  result = fcs_set_box_a(fcs, box_x);
+  if (!check_result(result, rank)) return;
+
+  result = fcs_set_box_b(fcs, box_y);
+  if (!check_result(result, rank)) return;
+
+  result = fcs_set_box_c(fcs, box_z);
+  if (!check_result(result, rank)) return;
+
+  result = fcs_set_box_origin(fcs, offset);
+  if (!check_result(result, rank)) return;
+
+  result = fcs_set_periodicity(fcs, periodicity);
+  if (!check_result(result, rank)) return; 
+
+  result = fcs_set_near_field_flag(fcs, near_field_flag);
+  if (!check_result(result, rank)) return;
+
+  result = fcs_set_total_particles(fcs, atom->natoms);
+  if (!check_result(result, rank)) return;
+}
+
+/* ----------------------------------------------------------------------
+    check if box parameters changed, requiring a new call to fcs_tune
+------------------------------------------------------------------------- */
+bool FixScafacos::box_has_changed()
+{
+  bool changed = false;
+
+  double n_periodicity[3];
+  double n_offset[3];
+  double n_box_x[3];
+  double n_box_y[3];
+  double n_box_z[3];
+
+  int n_total_particles;
+
+  // store periodicity
+  n_periodicity[0] = domain->xperiodic;
+  n_periodicity[1] = domain->yperiodic;
+  n_periodicity[2] = domain->zperiodic;
+
+  // store offset of the system
+  n_offset[0] = domain->boundary[0][0];
+  n_offset[1] = domain->boundary[1][0];
+  n_offset[2] = domain->boundary[2][0];
+
+  // calculate box vectors
+  n_box_x[0] = domain->prd[0];
+  n_box_x[1] = n_box_x[2] = 0.0;
+
+  n_box_y[1] = domain->prd[1];
+  n_box_y[0] = n_box_y[2] = 0.0;
+
+  n_box_z[2] = domain->prd[2];
+  n_box_z[1] = n_box_z[0] = 0.0;
+
+  n_total_particles = atom->natoms;
+
+  changed = changed ||
+              ( n_periodicity[0] != periodicity[0] ) ||
+              ( n_periodicity[1] != periodicity[1] ) ||
+              ( n_periodicity[2] != periodicity[2] ) ||
+              ( n_offset[0] != offset[0] ) ||
+              ( n_offset[1] != offset[1] ) ||
+              ( n_offset[2] != offset[2] ) ||
+              ( n_box_x[0] != box_x[0] ) ||
+              ( n_box_x[1] != box_x[1] ) ||
+              ( n_box_x[2] != box_x[2] ) ||
+              ( n_box_y[0] != box_y[0] ) ||
+              ( n_box_y[1] != box_y[1] ) ||
+              ( n_box_y[2] != box_y[2] ) ||
+              ( n_box_z[0] != box_z[0] ) ||
+              ( n_box_z[1] != box_z[1] ) ||
+              ( n_box_z[2] != box_z[2] ) ||
+              ( n_total_particles != total_particles );
+  return changed;
+}
+
+
+
+/* ----------------------------------------------------------------------
+    check of ScaFaCoS result
+------------------------------------------------------------------------- */
+bool FixScafacos::check_result(FCSResult result, int comm_rank) 
+{
+  if (result) 
+  {
+    printf("ScaFaCoS Error: Caught error on task %d.\n", comm_rank);
```

Comment:
Please don't use `printf()`, but `fprintf(screen,...)` and `fprintf(logfile, ...)` provided either FILE pointer is non-NULL. Please note, that calling error->all() requires, that this function is called from *all* MPI ranks on the communicator `world`, otherwise you need to call error->one(). If however `check_results()` *is* called from all MPI ranks on `world`, then only MPI rank 0 should be producing output from `fprintf()` or else the screen would have all messages replicated by the number of MPI ranks in use.